Tests/SourceKitTests/SwiftCompletionTests.swift

```diff
@@ -221,4 +227,30 @@ final class SwiftCompletionTests: XCTestCase {
       position: Position(line: 1, utf16index: 0)))
     XCTAssertTrue(outOfRange2.isIncomplete)
   }
+
+  func testCompletionOptional() {
```

Comment:
We should check a case where the deletion range is >1.  E.g. this will replace "func " with "override func foo() {}"
```swift
class Base {
  func foo() {}
}
class C: Base {
  func  <<<HERE>>>
}
```

Similarly, you should check what happens when this case crosses lines.  If sourcekitd deletes before the start of the line we should at least not crash here.
```swift
class C: Base {
  func 
  <<<HERE>>>
}
```

Reply:
Oh, good catch. I didn’t think of that case. I implemented the correct lookup logic using the snapshot now. This also fixes the UTF-8 to UTF-16 conversion issue. 

I expect this to be fine performance-wise since the extra conversions are only done when `utf8BytesToErase` is not 0 which I expect to happen fairly rarely and especially not on a huge amount of results.

Reply:
The most common case is "." to "?.", which will apply to all the results except those that come from `Optional` itself.  If it's an issue though we can special case `num_bytes_to_erase == 1`, since that can only ever be ASCII (so 1 UTF-16 code unit), or an invalid range that would currently trigger your assertion failure I believe.